mac/hid.c

```diff
@@ -696,12 +696,12 @@ hid_device * HID_API_EXPORT hid_open_path(const char *path)
 	io_registry_entry_t entry = MACH_PORT_NULL;
 	IOReturn ret = kIOReturnInvalid;
 
-	dev = new_hid_device();
-
 	/* Set up the HID Manager if it hasn't been done */
 	if (hid_init() < 0)
 		return NULL;
```

Comment:
Wouldn't it be better to use `goto return_error` instead of `return NULL`? It will solve the problem as well, but also will prevent any similar problems if anyone put any allocation before `if (hid_init() < 0)` line in the future.

Reply:
PR author is not answering, but I think the patch is more than acceptable as is - it does better then before in any case

Reply:
@Qbicz I'm not well versed enough in the library to have an opinion, hence why I didn't answer. I figured someone else might have an opinion on it.

Reply:
I agree with @Qbicz